ascent_macro/src/ascent_hir.rs

```diff
@@ -514,11 +515,15 @@ pub(crate) fn prog_get_relation<'a>(
          if rel.field_types.len() != arity {
             Err(Error::new(
                name.span(),
-               format!("Wrong arity for relation {}. Actual arity: {}", name, rel.field_types.len()),
+               format!(
+                  "wrong arity for relation `{name}` (expected {expected}, found {actual})",
+                  expected = arity,
+                  actual = rel.field_types.len()
+               ),
             ))
          } else {
             Ok(rel)
          },
-      None => Err(Error::new(name.span(), format!("Relation {} not defined", name))),
+      None => Err(Error::new(name.span(), format!("relation `{}` not defined", name))),
```

Comment:
Since you are changing this line, could you change the message to `"relation {} is not defined"`?

Reply:
Done.